src/fev/task.py

```diff
@@ -207,21 +267,66 @@ class Task(_TaskBase):
+    min_context_length: int = 1
+    max_context_length: int | None = None
+    # Evaluation parameters
+    seasonality: int = 1
+    eval_metric: str = "MASE"
+    extra_metrics: list[str] = dataclasses.field(default_factory=list)
+    quantile_levels: list[float] | None = None
+    # Feature information
+    id_column: str = "id"
+    timestamp_column: str = "timestamp"
+    target_column: str | list[str] = "target"
+    generate_univariate_targets_from: list[str] | Literal["__ALL__"] | None = None
+    past_dynamic_columns: list[str] = dataclasses.field(default_factory=list)
+    excluded_columns: list[str] | Literal["__ALL__"] = dataclasses.field(default_factory=list)
+    task_name: str | None = None
+
     def __post_init__(self):
-        super().__post_init__()
-        if self.generate_univariate_targets_from is not None and self.is_multivariate:
-            raise ValueError(
-                "`generate_univariate_targets_from` cannot be used for multivariate tasks (when `target_column` is a list)"
-            )
+        if self.task_name is None:
+            if self.dataset_config is not None:
+                # HF Hub dataset -> name of dataset_config
+                self.task_name = self.dataset_config
+            else:
+                # File dataset -> names of up to 2 parent directories
+                # e.g. /home/foo/bar/data.parquet -> foo/bar
+                self.task_name = "/".join(Path(self.dataset_path).parts[-3:-1])
 
-        if self.cutoff is None:
-            self.cutoff = -self.horizon
-        elif isinstance(self.cutoff, str):
-            self.cutoff = pd.Timestamp(self.cutoff).isoformat()
+        assert self.num_windows >= 1, "`num_windows` must satisfy >= 1"
+        if self.window_step_size is None:
+            self.window_step_size = self.horizon
+        if isinstance(self.window_step_size, int):
+            assert self.window_step_size >= 1, "`window_step_size` must satisfy >= 1"
+        else:
+            offset = pd.tseries.frequencies.to_offset(self.window_step_size)
+            assert offset.n >= 1, "If `window_step_size` is a string, it must correspond to a positive timedelta"
+            self.window_step_size = offset.freqstr
+
+        if self.initial_cutoff is None:
+            assert isinstance(self.window_step_size, int), (
+                "If `initial_cutoff` is None, `window_step_size` must be an int"
+            )
+            self.initial_cutoff = -self.horizon - (self.num_windows - 1) * self.window_step_size
+
+        if isinstance(self.initial_cutoff, int):
+            if not isinstance(self.window_step_size, int):
+                raise ValueError("`window_step_size` must be an int if `initial_cutoff` is an int")
+            assert self.window_step_size >= 1
+            max_allowed_cutoff = -self.horizon - (self.num_windows - 1) * self.window_step_size
+            if self.initial_cutoff < 0 and self.initial_cutoff > max_allowed_cutoff:
```

Comment:
I don't quite follow this check. So if I understand correctly I can only specify initial cutoff to allow for _at least_ the required number of steps (roughly horizon * nr_windows) in the future. if so, why is it not checked for timestamp as well?

Reply:
cool